scripts/api.js

```diff
@@ -35,29 +42,22 @@ function ajax(method, apiUrl, queryParams = {}, data = {}, fetchPromise = fetch)
         options.headers["Content-Type"] = "application/json";
         options.body = JSON.stringify(data);
     }
-    return new Promise(function (resolve, reject) {
-        fetchPromise(url, options)
-            .then(function (response) {
-                const contentType = response.headers.get("content-type");
-                if (!contentType || !contentType.includes("application/json")) {
-                    reject({ error: "Incorrect response type", code: AJAX_ERROR_CODES.INCORRECT_RESPONSE_TYPE });
-                } else if (response.ok) {
-                    resolve(response.json());
-                } else {
-                    response.json().then(function (data) {
-                        if (!data) {
-                            data = { error: "Unknown error", code: AJAX_ERROR_CODES.UNKNOWN_ERROR };
-                        }
-                        reject(data);
-                    });
-                }
-            })
-            .catch(function () {
-                reject({ error: "Network error", code: AJAX_ERROR_CODES.NETWORK_ERROR });
-            });
-    });
-}
-
-function ajaxAlert(data) {
-    alert(data.error);
+    let response;
+    try {
+        response = await fetchPromise(url, options);
+    } catch (error) {
+        throw new AjaxError(error, AJAX_ERROR_CODES.NETWORK_ERROR);
+    }
+    const contentType = response.headers.get("content-type");
+    if (!contentType || !contentType.includes("application/json")) {
+        throw new AjaxError("Incorrect response type", AJAX_ERROR_CODES.INCORRECT_RESPONSE_TYPE);
+    }
+    const responseData = await response.json();
```

Comment:
Note .json can throw, not sure if we want to handle that.

Reply:
Good point. I'll handle it and add a test.